src/scripts/main.js

```diff
@@ -1,11 +1,25 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener(
+      eventName,
+      (e) => {
+        resolve(
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        );
+      },
+      { once: true },
+    );
+  });
```

Comment:
It seems that the `reject` function of the Promise is never used. While it's not mandatory to use it in every Promise, consider whether there might be cases where the Promise should be rejected, such as if the event listener fails to be added or if there is a timeout waiting for the event.